docs/advanced_usages/custom-tests.md

```diff
@@ -0,0 +1,218 @@
+    * result = "failure" otherwise.
+    * result = "error" if any exception is caught
+"""
+```
+
+You can see output using [mkdocstring-python](https://mkdocstrings.github.io/python/) here: [tests.software.md](../api/tests.software.md)
+
+### Check inputs
+
+If your test has some user's inputs, first you have to validate values are valid. If it is not valid, we expect `TestResult` to return `skipped` with a custom message
+
+```python
+# Check if test option is correct
+if not minimum:
+    result.is_skipped("verify_dynamic_vlan was run without minimum value set")
+    return result
+```
+
+### Implement your logic
+
+Here you implement your own logic. In general, the first action is to send command to devices and capture its response.
+
+In the example below, we request list of vlans configured on device and then count all the vlans marked as dynamic
+
+```python
+# Send command to device
+response = await device.session.cli(command="show vlan", ofmt="json")
+
+# Do your test: In this example we count number of vlans with field dynamic set to true
+num_dyn_vlan = len([ vlan for vlan,data in response['vlans'].items() if data['dynamic'] is True])
+if num_dyn_vlan >= minimum:
+    result.is_success()
+else:
+    result.is_failure(f"Device has {num_dyn_vlan} configured, we expect at least {minimum}")
+
+return result
+```
+
+As you can see there is no error management to do in your code. Everything is packaged in `anta_tests` and below is a simple example of error captured with an incorrect JSON key in the code above:
+
+```bash
+ERROR    Exception raised for test verify_dynamic_vlan (on device 192.168.0.10) - KeyError ('vlans')
+```
+
+!!! info "Get stack trace for debugging"
+    If you want to access to the full exception stack, you can run your test with logging level set to `DEBUG`. With ANTA cli, it is available with following option:
+    ```bash
+    $ anta nrfu text --catalog test_custom.yml --log-level debug
+    ```
+
+## Create your catalog
+
+It is very similar to what is documented in [catalog section](../usage-inventory-catalog.md) but you have to use your own package name.
+
+Let say my custom catalog is `anta_titom73` and my test is configured in `anta_titom73.dc_project`, my test catalog should be:
+
+```yaml
+anta_titom73.dc_project:
+  - verify_dynamic_vlan:
+      minimum: 1
+```
+And now you can run your NRFU tests with the CLI:
+
+```bash
+anta nrfu text --catalog test_custom.yml
+spine01 :: verify_dynamic_vlan :: FAILURE (Device has 0 configured, we expect at least 1)
+spine02 :: verify_dynamic_vlan :: FAILURE (Device has 0 configured, we expect at least 1)
+leaf01 :: verify_dynamic_vlan :: SUCCESS
+leaf02 :: verify_dynamic_vlan :: SUCCESS
+leaf03 :: verify_dynamic_vlan :: SUCCESS
+leaf04 :: verify_dynamic_vlan :: SUCCESS
+```
+
+!!! warning "Install your python package"
+    Anta uses Python path to access to your test. So it is critical to have your tests library installed correctly.
```

Comment:
or simply in the PATH?

Reply:
In general, it is under `$PYTHONPATH` (`sys.path`) that is different than `$PATH`

```bash
python
Python 3.9.9 (main, Jan 26 2022, 14:06:08)
[Clang 12.0.5 (clang-1205.0.22.11)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.path
['', '/Users/tgrimonet/.pyenv/versions/3.9.9/lib/python39.zip', '/Users/tgrimonet/.pyenv/versions/3.9.9/lib/python3.9', '/Users/tgrimonet/.pyenv/versions/3.9.9/lib/python3.9/lib-dynload', '/Users/tgrimonet/.pyenv/versions/act-lab/lib/python3.9/site-packages']
>>> exit()

❯ echo $PATH
/Users/tgrimonet/.pyenv/plugins/pyenv-virtualenv/shims:/Users/tgrimonet/.pyenv/shims:/Users/tgrimonet/.pyenv/plugins/pyenv-virtualenv/shims:/Users/tgrimonet/.pyenv/bin:/Users/tgrimonet/.pyenv/plugins/pyenv-virtualenv/shims:/Users/tgrimonet/bin:/Users/tgrimonet/.local/bin:/usr/local/bin:/usr/sbin:/usr/local/sbin:/Users/tgrimonet/.pyenv/shims:/Users/tgrimonet/.pyenv/bin:/Users/tgrimonet/.pyenv/shims:/Users/tgrimonet/.pyenv/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/Applications/Wireshark.app/Contents/MacOS:/usr/local/go/bin:/Users/tgrimonet/go/bin:/Users/tgrimonet/.fzf/bin
```

Reply:
yeah sure used the wrong name but I meant maybe you don't need to have your library installed
you can just do

```
PYTHONPATH=$PYTHONPATH:<path/to/my/custom/package> anta ...
```

and it will work

Reply:
Sure any hack is doable here. I prefer to stick to python best practices and anyone can adapt if they want something else :)

Maybe we should add another disclaimer to state we are not dealing with how to make package available in Python (just how to create it) ?

Reply:
I would argue that making a package available in PYTHONPATH is not a hack as the recommendation to install it is just to achieve this.

Typically making a package installable means having to write some fluffy stuff to make it installable (setup,py, pyproject.toml, ...) which is some useless overhead if you just want to have a collection of custom tests locally whereas just writing a piece of code that is available in your PYTHONPATH to be loaded by anta is more straightforward in my opinion.

I would just state that the module with the custom states must be importable by anta and hence available on the PYTHONPATH

Reply:
Ok, so I reword a bit in the first section in this commit 7a063242bb57d99deed66453f6381511f6b7bfee